spanner/google/cloud/spanner_v1/_helpers.py

```diff
@@ -19,80 +19,12 @@
 
 import six
 
-from google.protobuf.struct_pb2 import ListValue
-from google.protobuf.struct_pb2 import Value
-from google.cloud.spanner_v1.proto import type_pb2
-
+from google.api_core import datetime_helpers
 from google.cloud._helpers import _date_from_iso8601_date
 from google.cloud._helpers import _datetime_to_rfc3339
-from google.cloud._helpers import _RFC3339_NANOS
-from google.cloud._helpers import _RFC3339_NO_FRACTION
-from google.cloud._helpers import UTC
-
-
-class TimestampWithNanoseconds(datetime.datetime):
-    """Track nanosecond in addition to normal datetime attrs.
-
-    nanosecond can be passed only as a keyword argument.
-    """
-    __slots__ = ('_nanosecond',)
-
-    # pylint: disable=arguments-differ
-    def __new__(cls, *args, **kw):
-        nanos = kw.pop('nanosecond', 0)
-        if nanos > 0:
-            if 'microsecond' in kw:
-                raise TypeError(
-                    "Specify only one of 'microsecond' or 'nanosecond'")
-            kw['microsecond'] = nanos // 1000
-        inst = datetime.datetime.__new__(cls, *args, **kw)
-        inst._nanosecond = nanos or 0
-        return inst
-    # pylint: disable=arguments-differ
-
-    @property
-    def nanosecond(self):
-        """Read-only: nanosecond precision."""
-        return self._nanosecond
-
-    def rfc3339(self):
-        """RFC 3339-compliant timestamp.
-
-        :rtype: str
-        :returns: Timestamp string according to RFC 3339 spec.
-        """
-        if self._nanosecond == 0:
-            return _datetime_to_rfc3339(self)
-        nanos = str(self._nanosecond).rstrip('0')
-        return '%s.%sZ' % (self.strftime(_RFC3339_NO_FRACTION), nanos)
-
-    @classmethod
-    def from_rfc3339(cls, stamp):
-        """Parse RFC 3339-compliant timestamp, preserving nanoseconds.
-
-        :type stamp: str
-        :param stamp: RFC 3339 stamp, with up to nanosecond precision
-
-        :rtype: :class:`TimestampWithNanoseconds`
-        :returns: an instance matching the timestamp string
-        :raises ValueError: if ``stamp`` does not match the expected format
-        """
-        with_nanos = _RFC3339_NANOS.match(stamp)
-        if with_nanos is None:
-            raise ValueError(
-                'Timestamp: %r, does not match pattern: %r' % (
-                    stamp, _RFC3339_NANOS.pattern))
-        bare = datetime.datetime.strptime(
-            with_nanos.group('no_fraction'), _RFC3339_NO_FRACTION)
-        fraction = with_nanos.group('nanos')
-        if fraction is None:
-            nanos = 0
-        else:
-            scale = 9 - len(fraction)
-            nanos = int(fraction) * (10 ** scale)
-        return cls(bare.year, bare.month, bare.day,
-                   bare.hour, bare.minute, bare.second,
-                   nanosecond=nanos, tzinfo=UTC)
+from google.cloud.spanner_v1.proto import type_pb2
+from google.protobuf.struct_pb2 import ListValue
+from google.protobuf.struct_pb2 import Value
```

Comment:
Per [PEP 8](https://www.python.org/dev/peps/pep-0008/#id23), non-local imports (e.g., `google.protobuf`) are supposed to be sorted between standard-library imports and local imports.